docs/DeveloperGuide.adoc

```diff
@@ -111,156 +111,369 @@ image::DeleteSequenceDiagram.png[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+==== Current Implementation
 
-_{Explain here how the data encryption feature will be implemented}_
+The state of the GUI is stored in a `GuiState` object. The `GuiState` object contains a `TabManager` object, which manages tab information such as the tabs that are open and the tabs that are currently selected.
 
-// end::dataencryption[]
+The information is linked to the `UI` through listeners. When the attributes of the `TabManager` object is updated, it triggers callback functions in the UI controllers that update the state of the UI displayed.
 
-=== Logging
+The following commands facilitate the management of tabs:
 
-We are using `java.util.logging` package for logging. The `LogsCenter` class is used to manage the logging levels and logging destinations.
+* `AlgoBase#switchTab()` - Switch between tabs within a specified Tab pane.
+* `AlgoBase#openTab()` - Opens a new tab containing details of a model.
+* `AlgoBase#closeTab()` - Closes an existing tab.
 
-* The logging level can be controlled using the `logLevel` setting in the configuration file (See <<Implementation-Configuration>>)
-* The `Logger` for a class can be obtained using `LogsCenter.getLogger(Class)` which will log messages according to the specified logging level
-* Currently log messages are output through: `Console` and to a `.log` file.
+These operations are exposed in the `TabManager` class respectively as:
 
-*Logging Levels*
+* `SwitchTab()`: `TabManager#setDisplayTabIndex` and `TabManager#setDetailsTabIndex`
+* `OpenTab()`: `TabManager#addTab`
+* `CloseTab()`: `TabManager#removeTab`
+
+The following Activity Diagrams illustrate what happens when the user executes a `SwitchTabCommand` or `OpenTabCommand`
+
+image::gui/SwitchTabCommandActivityDiagram.png[]
+_Figure 1.1 Activity Diagram for the Execution of `switchtab` Command_
+
+image::gui/OpenTabCommandActivityDiagram.png[]
+_Figure 1.2 Activity Diagram for the Execution of `opentab` Command_
+
+Given below is an example usage scenario and how the tag mechanism behaves at each step.
+
+image::gui/SwitchTabsSequenceDiagram0.png[]
+_Figure 2.1 Sequence Diagram for instantiating a `SwitchCommand` object_
+
+Step 1: The user executes `switchtab t/display i/1` to switch to the first tab in the `display` tabpane.
+
+Step 2: `SwitchTabCommandParser` processes the user input, retrieving the tab type (`display`) and the index (`1`).
+
+Step 3: These two attributes are passed into the constructor of a `SwitchTabCommand` and a corresponding `SwitchTabCommand` object is returned to the LogicManager
+
+
+image::gui/SwitchTabsSequenceDiagram1.png[]
+_Figure 2.2 Sequence Diagram for updating the tab index in the TabManager_
+
```

Comment:
This diagram is for calling getTabManager from GuiState is kind of confusing and you may want to have a look.